homeassistant/components/amcrest/__init__.py

```diff
@@ -259,6 +263,7 @@ def setup(hass, config):
 
         discovery.load_platform(hass, CAMERA, DOMAIN, {CONF_NAME: name}, config)
 
+        event_codes = ()
```

Comment:
We set this variable to a list below, so we should keep the same type here.

Reply:
I was trying to avoid allocating a new list, but fair enough (empty lists are all different, whereas the empty tuple is the exact same object each time, thus no new allocation). A better way may to make event_codes a set at this point already, and extend it just below, but I wanted the lowest disturbance risk by keeping the changes as local as possible.